test_runner/fixtures/neon_fixtures.py

```diff
@@ -752,7 +753,7 @@ def cleanup_local_storage(self):
     def cleanup_remote_storage(self):
         # here wee check for true remote storage, no the local one
         # local cleanup is not needed after test because in ci all env will be destroyed anyway
-        if self.remote_storage_prefix is None:
+        if not isinstance(self.remote_storage, S3Storage) or self.remote_storage_prefix is None:
```

Comment:
Shouldn't this `isinstance` be an assertion? If `remote_storage` is not S3, something is probably off.

Reply:
We call `cleanup_remote_storage` unconditionally  `NeonEnvBuilder.__exit__`

Reply:
Actually, since `self.remote_storage_prefix` is coupled with `S3Storage`, I've moved to `isinstance(self.remote_storage, S3Storage)` to and assert